docs/providers/aws/events/apigateway.md

```diff
@@ -812,6 +812,47 @@ functions:
               application/json: ${file(create_request.json)}
 ```
 
+API Gateway models can be customized by inline properties
+
+```yml
+functions:
+  create:
+    handler: posts.create
+    events:
+      - http:
+          path: posts/create
+          method: post
+          requestSchema:
+            application/json:
+              schema: ${file(create_request.json)}
+              name: PostCreateModel
+              description: 'Validation model for Creating Posts'
+```
+
+API Gateway models can be references by global models in the provider. The same structure exist
+for models defined in the provider as inline functions. Provider models default to `application/json`
+
+```yml
+provider:
+    ...
+    apiGateway:
+      requestSchemas:
+        post-create-model:
+          name: PostCreateModel
+          schema: ${file(api_schema/post_add_schema.json)}
+          description: "A Model validation for adding posts"
+
+functions:
+  create:
+    handler: posts.create
+    events:
+      - http:
+          path: posts/create
+          method: post
+          requestSchema:
```

Comment:
While I was wrapping up the last changes (deprecation), I've started wondering about this approach to having `provider.apiGateway.request.schemas` and `http.request.schemas` and deprecating `http.request.schema` - while `provider.apiGateway.request.schemas` sounds good, I'm wondering if we should/need to move from `http.request.schema` to `http.request.schemas` - what is the main reason here? Is it to be consistent and use plural `schemas`? Otherwise, we could totally accomodate the old `http.request.schema` to support both new and old approach (because it's only extending it). Please let me know what do you think. :bow:

Reply:
> Is it to be consistent and use plural schemas?

Yes, in first it's to be consistent (it's a bit confusing to have `schemas` as global, and `schema` as local, with very same _definition_ underneath). Also I think plural form is right, as we're configuring dictionary of schemas. Singular name may suggest we're defining the single schema, and those are its properties, which is again confusing.

While I understand the _pain_ of creating deprecation just because we add `s` I feel there's good reasoning behind it, and that it'll leave our API more solid.

What do you think?

Reply:
Thanks for the explanation - that's what I thought here, but wanted to confirm as I wasn't 100% sure. I think it makes sense to be consistent here - the migration path from `schema` to `schemas` is not complicated so it shouldn't be too painful for users. Thanks for the clarification :100: